design/ceph/object/store.md

```diff
@@ -103,6 +104,14 @@ The RGW service can be configured to listen on both http and https by specifying
     instances: 1
 ```
 
+#### Hostname Configuration
+
+These hostnames will be added to the zonegroup's hostname list and objectstore can be accessed only by the hostname from the list. If the feature is enabled, users need to add the all the endpoints like the openshift route, ingress loadbalancer and k8s service endpoint created by Rook to the list. The hostname list can be updated at any time by updating the object store CRD. Even though settings related to `zonegroup` to make consistent for single and multisite use case it is added to the object store CRD. After creating the object store, the Rook operator will update the zonegroup to include the `hostnames` to its list via `radosgw-admin` command.
+
+This is different from `customEndpoints` which is used for configuring the object store to replicate and sync data amongst Multisite. The `customEndpoints` are not added to the zonegroup's hostname list.
+
+The user need to create `DNS entry` for the hostname appending wildcard `*` like `*.rook-ceph-rgw-ocs-storagecluster-cephobjectstore.openshift-storage.svc:80 IN CNAME rook-ceph-rgw-ocs-storagecluster-cephobjectstore.openshift-storage.svc:80`. Same is applicable for TLS certs otherwise buckets won't be able to accessed via this endpoint.
```

Comment:
I don't see how Rook can create DNS entries. We need to find a way to use the K8s services for this.

Reply:
Right. I wonder if this should be left up to users. I think Rook could probably do this for Services (Rook would likely need to expose some service config options to users), but I'm not sure if OpenShift users would prefer to use Routes to create endpoints. It seems like it might be easiest to start with a design where this is assumed to be done by users, and then we can look for opportunities to do it automatically with Rook once we get feedback. 

IMO, services are such a simple building block in K8s that once we start allowing users to configure them (via passthru options in CRDs), we've basically made it necessary for the users to define more than half of the service itself. Plus, Rook has to document how users can find the services it creates that correlate to each hostname. It seems like it could be a better experience for everyone if Rook allows users to just do that themselves:
- we don't need to plan our design around user corner cases like service type or Route usage
- users won't feel confused about what the service name will be for each config they input
- users can define their own specialized service names if they want to, that are likely easier for them to remember